docs/docs/secrets-managers.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: secrets-managers
+sidebar_label: Secrets Managers
+title: Secrets Managers
+description: Credentials used to authenticate to external services are safely stored in a secrets manager.
+abstract: Credentials managers are used to safeguard credentials used to authenticate to external services.
+---
+
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import RasaProLabel from '@theme/RasaProLabel';
+import RasaProBanner from '@theme/RasaProBanner';
+
+<RasaProLabel />
+<RasaProBanner/>
+
+## Secrets Managers
+Secrets managers are used to safeguard credentials used to authenticate to external services,
+like databases.
+
+Rasa Pro supports the following secrets managers:
+* [HashiCorp Vault](https://www.hashicorp.com/products/vault)
+
+
+Currently, Rasa Pro supports sefeguarding credentials for the following services:
+- [Tracker Stores](./tracker-stores.mdx)
+
+
+## Vault Secrets Manager
+
+The Vault Secrets Manager is used to store credentials used to authenticate to external services.
+The credentials are stored in a Vault instance and can be encrypted at rest.
+
+If you need to encrypt your credentials,
+you can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
+
+:::note
+Expiring tokens need to be renewed periodically, and renewal process is done over the network,
+we will try to renew the token 5 seconds before it expires.
+If token's TTL is less than 5 seconds, we will try to renew it ASAP,
+but it might fail due to network latency.
```

Comment:
I think we need a suggestion / recommendation here. What happens if it fails, what will users need to do? Do we recommend TTL > 5s ? What does HashiCorp recommend?

Reply:
> What happens if it fails, what will users need to do?
Users will have to create a new token and restart Rasa with that new token.

We can build re-try mechanism for renewal of tokens.

HashiCorp does not have general recommendation for Vault token's TTL.
We can say that we recommend that token's TTL to be at least 1 minute.

Reply:
I opened [PR](https://github.com/RasaHQ/rasa-plus/pull/372) to address this.

To accommodate for network latency I increased threshold for renewal so that it starts 15 seconds before token will expire and retry 5 times every second if renewal fails.